core/src/com/unciv/models/ruleset/unique/Unique.kt

```diff
@@ -49,19 +49,25 @@ class Unique(val text: String, val sourceObjectType: UniqueTarget? = null, val s
         condition: Unique,
         state: StateForConditionals
     ): Boolean {
+
+        val ruleset = state.civInfo?.gameInfo?.ruleSet
+
         return when (condition.type) {
             UniqueType.ConditionalWar -> state.civInfo?.isAtWar() == true
             UniqueType.ConditionalNotWar -> state.civInfo?.isAtWar() == false
-            UniqueType.ConditionalHappy -> 
+            UniqueType.ConditionalHappy ->
                 state.civInfo != null && state.civInfo.statsForNextTurn.happiness >= 0
             UniqueType.ConditionalGoldenAge ->
                 state.civInfo != null && state.civInfo.goldenAges.isGoldenAge()
             UniqueType.ConditionalBeforeEra ->
-                state.civInfo != null && state.civInfo.getEraNumber() < state.civInfo.gameInfo.ruleSet.eras[condition.params[0]]!!.eraNumber
+                state.civInfo != null && ruleset!!.eras.containsKey(condition.params[0])
```

Comment:
Shouldn't these kinds of checks be made when building the ruleset instead of during the game?

Reply:
Yes, which makes it all the stranger that the Google Play Console shows this isn't the case:

```
java.lang.NullPointerException: 
  at com.unciv.models.ruleset.unique.Unique.conditionalApplies (Unique.kt:62)
  at com.unciv.models.ruleset.unique.Unique.conditionalsApply (Unique.kt:43)
  at com.unciv.logic.civilization.CivilizationInfo$getMatchingUniques$2.invoke (CivilizationInfo.kt:405)
  at com.unciv.logic.civilization.CivilizationInfo$getMatchingUniques$2.invoke (CivilizationInfo.kt:404)
  at kotlin.sequences.FilteringSequence$iterator$1.calcNext (Sequences.kt:171)
  at kotlin.sequences.FilteringSequence$iterator$1.hasNext (Sequences.kt:194)
  at kotlin.sequences.FlatteningSequence$iterator$1.ensureItemIterator (Sequences.kt:316)
  at kotlin.sequences.FlatteningSequence$iterator$1.hasNext (Sequences.kt:303)
  at com.unciv.models.ruleset.unit.BaseUnit.canBePurchasedWithStat (BaseUnit.kt:705)
  at com.unciv.logic.city.INonPerpetualConstruction$DefaultImpls.canBePurchasedWithAnyStat (IConstruction.kt:51)
  at com.unciv.models.ruleset.unit.BaseUnit.canBePurchasedWithAnyStat (BaseUnit.kt:26)
  at com.unciv.models.ruleset.unit.BaseUnit.shouldBeDisplayed (BaseUnit.kt:371)
  at com.unciv.ui.cityscreen.CityConstructionsTable$getConstructionButtonDTOs$1.invoke (CityConstructionsTable.kt:166)
  at com.unciv.ui.cityscreen.CityConstructionsTable$getConstructionButtonDTOs$1.invoke (CityConstructionsTable.kt:166)
  at kotlin.sequences.FilteringSequence$iterator$1.calcNext (Sequences.kt:171)
  at kotlin.sequences.FilteringSequence$iterator$1.hasNext (Sequences.kt:194)
  at com.unciv.ui.cityscreen.CityConstructionsTable.getConstructionButtonDTOs (CityConstructionsTable.kt:166)
  at com.unciv.ui.cityscreen.CityConstructionsTable.access$getConstructionButtonDTOs (CityConstructionsTable.kt:30)
  at com.unciv.ui.cityscreen.CityConstructionsTable$updateAvailableConstructions$1.invoke (CityConstructionsTable.kt:208)
  at com.unciv.ui.cityscreen.CityConstructionsTable$updateAvailableConstructions$1.invoke (CityConstructionsTable.kt:206)
  at kotlin.concurrent.ThreadsKt$thread$thread$1.run (Thread.kt:30)
```

Reply:
If I'm not mistaken, we don't actually check the validity of parameters of conditionals yet, which would also explain the crash

Reply:
Oh.
Actually it appears we don't check this.
Ruleset.checkUniques runs unique.type.getComplianceErrors
But you know what doesn't run? 
unique.conditionals